node/src/components/block_proposer.rs

```diff
@@ -380,33 +382,74 @@ impl BlockProposerReady {
     ) -> ProtoBlock {
         let max_transfers = deploy_config.block_max_transfer_count as usize;
         let max_deploys = deploy_config.block_max_deploy_count as usize;
+        let max_block_size_bytes = deploy_config.max_block_size as usize;
+        let block_gas_limit = Gas::from(deploy_config.block_gas_limit);
 
         let mut transfers = Vec::new();
         let mut wasm_deploys = Vec::new();
+        let mut block_gas_running_total = Gas::zero();
+        let mut block_size_running_total = 0usize;
 
         for (hash, deploy_type) in self.sets.pending.iter() {
-            let under_max_transfers = transfers.len() < max_transfers;
-            let under_max_deploys = wasm_deploys.len() < max_deploys;
-            if !under_max_deploys && !under_max_transfers {
+            let at_max_transfers = transfers.len() == max_transfers;
```

Comment:
(Not necessarily in this PR, but:)
I thought we don't actually want a `max_transfers`, only a `max_deploys`. In other words: We want to reserve space for transfers, and use the remaining space for both transfers and deploys. (@AlexanderLimonov?)

Reply:
Regarding your last sentence - yes. I'm a bit confused about the limits imposed here being in terms of #s rather than gas (though, for transfers, it doesn't make a practical difference, only a difference for readability).

Reply:
So I guess this code will be updated accordingly when [CEP 22](https://github.com/CasperLabs/ceps/pull/22) gets implemented?